source/server/listener_manager_impl.h

```diff
@@ -36,9 +36,11 @@ class ProdListenerComponentFactory : public ListenerComponentFactory,
   }
   Network::ListenSocketPtr createListenSocket(Network::Address::InstanceConstSharedPtr address,
                                               bool bind_to_port) override;
+  uint64_t nextOpaqueId() override { return next_opaque_id_++; }
 
 private:
   Instance& server_;
+  uint64_t next_opaque_id_{};
```

Comment:
I wrote the code to not imply uniqueness because I don't think there is any reason to, even though it will be used that way right now. If you think otherwise let me know. Sure I can change to listenerTag() everywhere. Can also start with 1 for the factory.